Prevent polluting ENV with postgres vars

sj26 opened the pull request:
### Summary

Tasks like `db:setup` and `db:test:prepare` across multiple databases combined with postgresql databases in structure mode may fail due to environment variable pollution. The variables are used to target the `pg_dump` and `psql` commands, but persist beyond calling the commands and so can influence later commands and connections. This especially troublesome because not all variables are set for every connection configuration.

Passing the env to the commands explicitly instead should prevent pollution.

### Other Information

I'm still identifying the best place to add a test. Perhaps in the [postgres rake tests](https://github.com/rails/rails/blob/main/activerecord/test/cases/tasks/postgresql_rake_test.rb), checking that environment variables aren't present after the task and so haven't escaped?

sj26 committed:
Prevent polluting ENV with postgres vars

guilleiguaran commented:
Please update the related tests and add a CHANGELOG entry for this 👍

guilleiguaran reviewed (changes requested):
Please update the related tests and add a CHANGELOG entry for this 👍

sj26 commented:
Thanks @guilleiguaran, do you have any advice on the best way to test?

sj26 commented:
Oh maybe I can update the mocks and that will be enough.

guilleiguaran commented:
@sj26 yes, I think updating the mocks will be enough to get the test suite green.

sj26 commented:
I updated the broken mocks, and noticed that the environment variables which I've changed were untested, so I've added some new tests to make sure the environment variables are set as well.

sj26 commented:
.. and I've added a CHANGELOG entry. Please let me know if it is too verbose.

guilleiguaran commented:
@sj26 looks like the changelog entry is missing in the last commit

sj26 commented:
Sorry, I missed it in my last push! I've rebased on main to avoid a merge conflict on the CHANGELOG.

guilleiguaran commented:
Thank you very much!!!
